experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -433,33 +438,40 @@ export type StructTyped<TSchema extends StructSchema> = Struct &
 /**
  * Properties to access a struct nodes fields. See {@link StructTyped}.
  *
- * @privateRemarks
- * TODO: support custom field keys
+ * @privateRemarks TODOs:
+ *
+ * 1. Support custom field keys.
+ *
+ * 2. Do we keep assignment operator + "setFoo" methods, or just use methods?
+ * Inconsistency in the API experience could confusing for consumers.
  *
  * @alpha
  */
-export type StructFields<TFields extends RestrictiveReadonlyRecord<string, FieldSchema>> =
-	// Getters
-	{
-		readonly [key in keyof TFields]: UnboxField<TFields[key]>;
-	} & {
-		// boxed fields (TODO: maybe remove these when same as non-boxed version?)
-		readonly [key in keyof TFields as `boxed${Capitalize<key & string>}`]: TypedField<
-			TFields[key]
-		>;
-	};
-// TODO: Add `set` method when FieldKind provides a setter (and derive the type from it).
-// set(key: FieldKey, content: FlexibleFieldContent<TSchema["mapFields"]>): void;
-// {
-// 	readonly [key in keyof TFields as `set${Capitalize<key & string>}`]: (
-// 		content: FlexibleFieldContent<TFields[key]>,
-// 	) => void;
-// };
-// This could be enabled to allow assignment via `=` in some cases.
-// & {
-// 	// Setter properties (when the type system permits)
-// 	[key in keyof TFields]: UnwrappedField<TFields[key]> & StructSetContent<TFields[key]>;
-// }
+export type StructFields<TFields extends RestrictiveReadonlyRecord<string, FieldSchema>> = {
+	// boxed fields (TODO: maybe remove these when same as non-boxed version?)
+	readonly [key in keyof TFields as `boxed${Capitalize<key & string>}`]: TypedField<TFields[key]>;
+} & {
+	// Unboxed fields
+	readonly [key in keyof TFields]: UnboxField<TFields[key]>;
+} & {
+	// Add setter (make property writable) when schema type is a value.
+	// If we could map to getters and setters separately, we would preferably do that, but we can't.
```

Comment:
This is the TS issue to cite here: https://github.com/microsoft/TypeScript/issues/43826

Reply:
```suggestion
	// If we could map to getters and setters separately, we would preferably do that, but we can't.
	// See https://github.com/microsoft/TypeScript/issues/43826 for more details on this limitation.
```